src/sagemaker/estimator.py

```diff
@@ -614,16 +613,21 @@ def __init__(
         self.output_kms_key = resolve_value_from_config(
             output_kms_key, TRAINING_JOB_KMS_KEY_ID_PATH, sagemaker_session=self.sagemaker_session
         )
+        use_volume_kms_config: bool = False
         if instance_type is None or isinstance(instance_type, str):
             instance_type_for_volume_kms = instance_type
-        elif isinstance(instance_type, ParameterString):
-            instance_type_for_volume_kms = instance_type.default_value
+        elif isinstance(instance_type, PipelineVariable):
+            use_volume_kms_config = True
+            instance_type_for_volume_kms = instance_type
```

Comment:
Can we add a warning here? See previous comment https://github.com/aws/sagemaker-python-sdk/pull/4065#discussion_r1291621674

Reply:
Discussed offline, we will not include a comment to adhere to existing code style.